src/frontend/src/binder/relation/join.rs

```diff
@@ -28,6 +28,7 @@ pub struct BoundJoin {
     pub left: Relation,
     pub right: Relation,
     pub cond: ExprImpl,
+    pub is_lookup_join: bool,
```

Comment:
The choice of lookup join should left to planner, e.g. converting logical join node to batch join node, rather in binder.